ts/main.ts

```diff
@@ -219,18 +230,27 @@ function findExerciseByBaseId(baseId: number): Exercise | null {
       return exercise;
     }
   }
+  for (const exercise of fitlogData.favorites) {
+    if (exercise.baseId === baseId) {
+      return exercise;
+    }
+  }
   return null;
 }
 
 function handleFavoriteClick(
   exerciseObj: Exercise,
   targetIcon: HTMLElement,
 ): void {
+  $favoritesNodeList = document.querySelectorAll(
+    '#favorites-card-list > .card',
+  );
   if (targetIcon.classList.contains('fa-regular')) {
```

Comment:
As part of the above above comment, I would just make this a const variable and declare it here because you're only using it in this function.

Same thing with the exercises nodeList if you need it.